core/trino-main/src/main/java/io/trino/sql/planner/FragmentTableScanCounter.java

```diff
@@ -13,18 +13,22 @@
  */
 package io.trino.sql.planner;
 
+import com.google.common.base.Predicates;
+import com.google.common.collect.ImmutableSet;
+import io.trino.spi.connector.CatalogHandle;
 import io.trino.sql.planner.plan.ExchangeNode;
 import io.trino.sql.planner.plan.PlanNode;
 import io.trino.sql.planner.plan.PlanVisitor;
 import io.trino.sql.planner.plan.TableScanNode;
 
+import java.util.HashSet;
 import java.util.List;
+import java.util.Set;
+import java.util.function.Predicate;
 
 /**
  * Visitor to count number of tables scanned in the current fragment
  * (fragments separated by ExchangeNodes).
- * <p>
- * TODO: remove this class after we make colocated_join always true
  */
```

Comment:
this class is named `FragmentTableScanCounter`, so it not right name for `collectRemoteExchangeNodes`. Consider ranaming it